flank-scripts/src/main/kotlin/flank/scripts/integration/CommitList.kt

```diff
@@ -0,0 +1,34 @@
+@file:Suppress("EXPERIMENTAL_API_USAGE")
+
+package flank.scripts.integration
+
+import com.github.kittinunf.result.getOrElse
+import com.github.kittinunf.result.onError
+import flank.scripts.github.getGitHubCommitList
+import flank.scripts.github.getPrDetailsByCommit
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.flatMapMerge
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+
+suspend fun getCommitListSinceDate(token: String, since: String) = coroutineScope {
+    getGitHubCommitList(token, listOf("since" to since))
+        .onError { println(it.message) }
+        .getOrElse { emptyList() }
+        .asFlow()
+        .flatMapMerge {
+            channelFlow {
+                launch { send(it.sha to getPrDetailsByCommit(it.sha, token).get()) }
+            }
+        }
```

Comment:
Personally I do not think that we need flow here :) I know that they are cool, but here we could achive the same with kotlin map

Reply:
With a map, we would have 100% synchronous code. I think it's always a good idea to have HTTP requests implemented asynchronously (if possible).
On the other hand, I agree with you, this is rather simple logic, not many requests expected here.

I don't have a strong opinion, just wanted to have asynchronous calls (and play with flow 😄 ).
Let's wait a bit and if there are be more concerns I'll change it to  `map`

Reply:
I agree 100% with you that we should not do this in parallel, but I think that:

`.asFlow().map { } `will do the job 
or
`async { it.sha to getPrDetailsByCommit(it.sha, token).get()) }` will also do it

Reply:
I am afraid not, the commit list is emitted synchronously therefore each commit will walk through all the steps before the next one is processed. `channelFlow` actually creates a new flow with values emitted (sent) from coroutines

Reply:
`.asFlow()` -> will emit items
`.map {} `-> will handle them asynchronously

Reply:
https://github.com/Flank/flank/pull/1353#discussion_r540833375